lib/Onyx.js

```diff
@@ -624,9 +637,21 @@ function merge(key, val) {
 
 /**
  * Merge user provided default key value pairs.
+ *
+ * @returns {Promise}
  */
 function initializeWithDefaultKeyStates() {
-    _.each(defaultKeyStates, (state, key) => merge(key, state));
+    return AsyncStorage.multiGet(_.keys(defaultKeyStates))
+        .then((pairs) => {
+            const asObject = _.chain(pairs)
+                .map(([key, val]) => [key, val && JSON.parse(val)])
+                .object()
+                .value();
+
+            const merged = lodashMerge(asObject, defaultKeyStates);
+            cache.merge(merged);
+            _.each(merged, (val, key) => keyChanged(key, val));
```

Comment:
We don't need to write back to storage as `defaultKeyStates` is something that is always overwritten during init
Another reason to preserve in cache forever is that the constant `defaultKeyStates` is never cleared, which means these keys were already in memory for the entire life of the app